src/lib.rs

```diff
@@ -51,16 +86,25 @@ mod sys;
 #[path = "linux.rs"]
 mod sys;
 
-#[cfg(target_env = "sgx")]
-#[path = "sgx.rs"]
+#[cfg(not(any(target_os = "linux", target_os = "macos", target_arch = "wasm32")))]
+#[path = "dummy.rs"]
 mod sys;
 
-impl Dlmalloc {
-    /// Creates a new instance of an allocator, same as `DLMALLOC_INIT`.
-    pub fn new() -> Dlmalloc {
-        DLMALLOC_INIT
+impl Dlmalloc<System> {
+    /// Creates a new instance of an allocator
+    pub const fn new() -> Dlmalloc<System> {
+        Dlmalloc(dlmalloc::Dlmalloc::new(System::new()))
     }
+}
+
+impl<S> Dlmalloc<S> {
```

Comment:
I think that this and other type parameters may be best written as `A` after the rename?